src/page/sinfo.rs

```diff
@@ -32,15 +32,31 @@ impl core::fmt::Display for SecInfo {
         match self.class {
             Class::Secs => write!(f, "S"),
             Class::Tcs => write!(f, "T"),
+            // FIXME: Reg -> Regular
             Class::Reg => write!(f, "{}", self.flags),
+            // FIXME: Va -> VersionArray
             Class::Va => write!(f, "V"),
+            // FIXME: Trim -> Trimmed
             Class::Trim => write!(f, "^"),
+            Class::ShadowStackFirst => write!(f, "SSF"),
+            Class::ShadowStackRest => write!(f, "SSR"),
         }
     }
 }
 
 impl SecInfo {
+    /// Create a new instance.
+    #[inline]
+    pub const fn new(class: Class, flags: Flags) -> SecInfo {
+        Self {
+            class,
+            flags,
+            reserved: [0; 31],
+        }
+    }
```

Comment:
I think we should do this too:

```rust
impl From<Class> for SecInfo {
    fn from(class: Class) -> Self {
        Self { class, flags: Flags::empty(), reserved: [0; 31] }
    }
}
```

This allows the trivial creation of `SecInfo` from `Class::Tcs`. Perhaps you can even conditionalize the default value of `flags` depending on the `Class`. For example, `Class::Tcs` can default to no flags and `Class:Reg` could default to RW.

Reply:
So in this I chose based on hardware constraints:
- `Class::Tcs` -> RW
- `Class::Reg` -> R